### ISSUE TITLE
Google Pub/Sub sample could use love

### ISSUE THREAD

**Issue by mattmoor** [knative/docs#548]
https://github.com/knative/docs/tree/master/eventing/samples/gcp-pubsub-source#gcp-cloud-pubsub---source

In particular, why don't we just provide gcloud commands for step `#6` instead of having two sections with 4 sub-steps a piece.

**Comment by mattmoor** [comment]
cc @evankanderson @vaikas-google

**Comment by mattmoor** [comment]
`ko` doesn't support a `-n` option: https://github.com/knative/docs/tree/master/eventing/samples/gcp-pubsub-source#subscriber

**Comment by mattmoor** [comment]
^^ These shouldn't reference `ko` at all, but should at least work.

**Comment by mattmoor** [comment]
There's a number of places I've seen installing the latest nightly (should be 0.2.x)

**Comment by mattmoor** [comment]
[This](https://github.com/knative/docs/blob/master/eventing/samples/gcp-pubsub-source/subscriber.yaml) should use the spec:
```yaml
spec:
  channel:
    apiVersion: eventing.knative.dev/v1alpha1
    kind: Channel
    name: qux-1
  subscriber:
    ref:
      apiVersion: serving.knative.dev/v1alpha1
      kind: Service
      name: message-dumper
```

**Comment by mattmoor** [comment]
The fact that the secrets live in different namespaces burned me as well. :(

If I didn't have to modify the command, I probably would have copy/pasted them rather than trying to recycle my command history...

**Comment by mattmoor** [comment]
Now I'm seeing the following in the logs for the `gcppubsub` deployment in the default namespace:
```
  caller:  "gcppubsub/adapter.go:79"   
  level:  "error"   
  logger:  "fallback"   
  msg:  "Failed to post message{error 25 0  Post http://qux-1-channel.default.svc.cluster.local/: dial tcp 10.31.242.225:80: i/o timeout}"   
  stacktrace:  "github.com/knative/eventing-sources/pkg/adapter/gcppubsub.(*Adapter).receiveMessage
	/go/src/github.com/knative/eventing-sources/pkg/adapter/gcppubsub/adapter.go:79
github.com/knative/eventing-sources/pkg/adapter/gcppubsub.(*Adapter).Start.func1
	/go/src/github.com/knative/eventing-sources/pkg/adapter/gcppubsub/adapter.go:68
github.com/knative/eventing-sources/vendor/cloud.google.com/go/pubsub.(*Subscription).receive.func3
	/go/src/github.com/knative/eventing-sources/vendor/cloud.google.com/go/pubsub/subscription.go:509"   
  ts:  1542413495.6293843   
```

**Comment by mattmoor** [comment]
I was finally able to get the sample working on a cluster with the mesh installed.

**Comment by mattmoor** [comment]
/assign @evankanderson 

Do you want to take a crack at simplifying these instructions?

I poached a few of the simpler changes, but I think several more changes would be useful:
1. Surface copy/paste-able `gcloud` commands for secrets using concrete filenames
1. I think we should have less "if you modified X" sections (or collapse those), these are otherwise fluff that make the instructions even more daunting.  IMO this sample should be *super* streamlined.
1. Eliminate `ko` requirement everywhere.
1. Update install sources to not use nightly, but the official release.
1. Note that Eventing requires the mesh (for 0.2).

**Comment by evankanderson** [comment]
Looks like much of this was addressed by #673.
